### DISCUSSION TITLE
Why could sktime 0.27.0 be in conflict with Pandas 2.2.1?

### DISCUSSION THREAD

**Discussion by matthewferler** [alan-turing-institute/sktime#6180] [Q&A] [answered]
I have a Dockerfile that uses pip3 and requirements.txt to install several things including sktime and pandas.  When I specify the combination of pandas==2.2.1 and sktime==0.27.0 there is a pip3 error.  I don't understand why because the sktime docs indicate that Pandas 2 has been supported since version [0.19.1](https://www.sktime.net/en/stable/changelog.html#version-0-19-1-2023-05-30).  The output from pip3 is as follows.  Any help to understand this issue is greatly appreciated.

INFO: pip is looking at multiple versions of sktime to determine which version is compatible with other requirements.  This could take a while.
ERROR: Cannot install -r /app/requirements.txt (line 5) and pandas==2.2.1 because these package versions have conflicting dependencies.

The conflict is caused by:
   The user requested pandas==2.2.1
   sktime 0.27.0 depends on pandas<2.2.0 and >=1.1

**Comment by fkiraly** [comment]
We manage an upper bound of `pandas`, since there are close dependencies on objects like the `ForecastingHorizon` on `pandas` indicese, and MINOR version releases of `pandas`, in the past, have repeatedly broken core functionality.

An update that makes `sktime` compatible with `pandas 2.2.X` will be available in `sktime 0.28.0`, scheduled for the next days.

Apologies, the `pandas` incompatibilities tend to be increasingly obscure, as they have to do with `freq` and multiples of exotic frequency strings and the like.

**Reply by fkiraly** [reply]
In case you are an appreciator of subtle horror, here is the respective fix: https://github.com/sktime/sktime/pull/6057

Generally, contributions that help us address various warnings from future deprecations early are much appreciated!

Also, the `ForecastingHorizon` will need a rework at some time in the future, that's also a mini-project where help is appreciated.

**Reply by matthewferler** [reply]
Thank you for the information.  It's a great library.  I look forward to 0.28.0.

**Comment by fkiraly** [comment, marked as answer]
FYI @matthewferler, `sktime 0.28.0` has been released and is compatible with `pandas 2.2.X`. Hopefully, no further issues arise from the changes in how `pandas` handles `freq`.

**Reply by matthewferler** [reply]
I pulled sktime 0.28.0 into my environment along with pandas 2.2.1 successfully.  Thank you for all of your hard work!

**Reply by fkiraly** [reply]
You're welcome!
(donations of course are also appreciated 😉 https://opencollective.com/sktime )
